pytest/tests/sanity/block_sync_archival.py

```diff
@@ -113,28 +121,61 @@ def main(argv: typing.Sequence[str]) -> None:
 
     with Cluster(min_block_production_delay=min_delay,
                  max_block_production_delay=max_delay) as cluster:
-        # Start the validator and wait for a few epoch’s worth of blocks to be
-        # generated.
+        # Start the validator and the first observer.  Wait until the observer
+        # synchronises a few epoch’s worth of blocks to be generated and then
+        # kill validator so no more blocks are generated.
         boot = cluster.start_node(0)
-        latest = utils.wait_for_blocks(boot,
-                                       target=target_height,
-                                       poll_interval=1)
-
-        # Start the observer node and wait for it to catch up with the chain
-        # state.
         fred = cluster.start_node(1)
         utils.wait_for_blocks(fred, target=target_height, poll_interval=1)
+        boot.kill()
+        latest = fred.get_latest_block()
+
+        # Start the second observer node and wait for it to catch up with the
+        # first observer.
+        barney = cluster.start_node(2, boot_node=1)
+        utils.wait_for_blocks(barney, target=latest.height, poll_interval=1)
 
         # Verify that observer got all the blocks.  Note that get_all_blocks
         # verifies that the node has full chain from head to genesis block.
-        boot_blocks = get_all_blocks(boot, head=latest)
         fred_blocks = get_all_blocks(fred, head=latest)
-        if boot_blocks != fred_blocks:
-            for a, b in zip(boot_blocks, fred_blocks):
+        barney_blocks = get_all_blocks(barney, head=latest)
+        if barney_blocks != fred_blocks:
+            for a, b in zip(fred_blocks, barney_blocks):
                 if a != b:
                     logger.error(f'{a} != {b}')
             assert False
 
+        # Get near_partial_encoded_chunk_request_processing_time metric
+        response = requests.get('http://{}:{}/metrics'.format(*fred.rpc_addr()))
```

Comment:
let's create a util function to fetch prometheus metrics (this might be useful in other tests too)

Reply:
Let’s do that in follow up change. There’s already libs/metrics.py but it’s limited in what it’s doing so this needs a bit more work to be done properly.

Reply:
https://github.com/near/nearcore/issues/6458